src/stores/Store.ts

```diff
@@ -74,6 +78,184 @@ export default class Store {
+		this.editedMessageSendDelay = delay;
+	};
+
+	@observable setEditedMessageIndex = (index: number) => {
+		this.editedMessageIndex = index;
+		if (localStorage.getItem('editedMessageIndex') !== index.toString()) {
+			localStorage.setItem('editedMessageIndex', index.toString());
+		}
+	};
+
+	@action saveEditedMessage = () => {
+		if (this.editedMessageIndex >= 0) {
+			const editedMessage: string = JSON.stringify({
+				session: this.selectedSession,
+				dictionary: this.selectedDictionaryName,
+				messageType: this.selectedMessageType,
+				message: JSON.parse(this.parsedMessagesHistory[this.editedMessageIndex]).message,
+				delay: this.editedMessageSendDelay,
+			});
+			this.parsedMessagesHistory[this.editedMessageIndex] = editedMessage;
+			this.indicators[this.editedMessageIndex] = 'circleEdited';
+			this.setEditedMessageIndex(-1);
+			this.setEditMessageMode(false);
+		}
+	};
+
+	@action setEditMessageMode=(mode: boolean) => {
+		this.editMessageMode = mode;
+		if (mode === false) {
+			this.setSelectedSession(null);
+			this.setSelectedMessageType(null);
+			this.setDictionaryName(null);
+			this.setEditedMessageIndex(-1);
+			this.setSendButtonIcon('arrow-right-icon');
+			this.setSendButtonTitle('Send Message');
+		}
+
+		if (!localStorage.getItem('editMessageMode') || localStorage.getItem('editMessageMode') !== mode.toString()) {
+			localStorage.setItem('editMessageMode', mode.toString());
+		}
+	};
+
+	@action selectMessage = (index: number) => {
+		this.setSelectedSession(JSON.parse(this.parsedMessagesHistory[index]).session);
+		this.setDictionaryName(JSON.parse(this.parsedMessagesHistory[index]).dictionary);
+		this.setSelectedMessageType(JSON.parse(this.parsedMessagesHistory[index]).messageType);
+		this.setEditorCode('{}');
+		this.setSendButtonTitle('Save');
+		this.setSendButtonIcon('');
+		this.setEditMessageMode(true);
+		this.setEditedMessageIndex(index);
+	};
+
+	@action setSendButtonIcon =(iconClass: string) => {
+		this.sendButtonIcon = iconClass;
+	};
+
+	@action setSendButtonTitle = (title: string) => {
+		this.sendButtonTitle = title;
+	};
+
+	@action setSelectedSession = (session: string | null) => {
+		this.selectedSession = session;
+		if (this.editMessageMode === false && session !== null) {
+			localStorage.setItem('selectedSession', session);
+		}
+	};
+
+	@action setSelectedMessageType = (messageType: string | null) => {
+		this.selectedMessageType = messageType;
+		if (this.editMessageMode === false && messageType !== null) {
+			localStorage.setItem('selectedMessageType', messageType);
+		}
+	};
+
+	@action setDictionaryName = (dictionary: string | null) => {
+		this.selectedDictionaryName = dictionary;
+		if (this.editMessageMode === false && dictionary !== null) {
+			localStorage.setItem('selectedDictionary', dictionary);
+		}
+	};
+
+	@action setEditorCode = (code: string) => {
+		this.editorCode = code;
+	};
+
+	@action getIndicatorByResult = (code: number): string => {
```

Comment:
Action suppose to modify state, this function so small and used only once, it doesnt even worth it to make it a function